peer.go

```diff
@@ -602,6 +604,11 @@ func (p *peer) WaitForDisconnect(ready chan struct{}) {
 	}
 
 	p.wg.Wait()
+
+	// Now that we are certain all active goroutines which could have been
+	// modifying the write buffer have exited, return the buffer to the pool
+	// to be reused.
+	p.server.writeBufferPool.Return(p.writeBuf)
```

Comment:
Alternatively, the peer that take the buffer _each_ time it goes to write a message. As is, would seem that we still need to commit 65KB to each active peer, with most of them likely being idle other than pings (once the inv system is set up). If they take the buffer each time they go to write, then eventually we'll enter a steady state of the total number of write bufs necessary to handle pings, and the links that are actually active. This'll likely have a _slight_ impact on performance, but when we get there I wager there're other lower hanging fruits to optimize away in the forwarding critical path.

Reply:
Re-reading the PR description, as is this would indeed still help with the allocations bursts with flappy peers, what I described above whoever would tend us to a minimal working set w.r.t the number of write buffers. Since the package is also nicely abstracted, we can eventually utilize this both within _lnwire_ and also _channeldb_ for serder purposes.

Reply:
Indeed we can go further, though I think that would require a little more research into the access patterns, which would inform us of how frequently we should return buffers to the pool. I'm not sure the performance penalty of doing it for every message is necessarily worth it, as traffic can often be bursty and we might send a few thousand messages at once.

One way I've though of doing this is to have a timer, that returns the buffer if no messages have been written in the last n seconds. Another alternative would be to allow the write handler to inspect the queue size, and return it if it ever goes to 0. I think both would work, but a little more involved. I agree we should first ensure our memory usage is bounded by the number of active peers, and then follow up to make it bounded by number of concurrent messages.